packages/web-components/fast-foundation/src/select/select.ts

```diff
@@ -0,0 +1,225 @@
+     *
+     * @public
+     */
+    @volatile
+    public get displayValue(): string {
+        if (this.firstSelectedOption) {
+            return this.firstSelectedOption.textContent || this.firstSelectedOption.value;
+        }
+        return "";
+    }
+
+    /**
+     * Sets the value when the options are changed.
+     *
+     * @param prev - The previous value
+     * @param next - the new value
+     * @internal
+     */
+    protected selectedOptionsChanged(prev: Option[] = [], next: Option[] = []): void {
+        super.selectedOptionsChanged(prev, next);
+        if (this.$fastController.isConnected) {
+            this.value = `${
+                this.firstSelectedOption.value || this.firstSelectedOption.textContent
+            }`;
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public clickHandler(e: MouseEvent): boolean | void {
+        // do nothing if the select is disabled
+        if (this.disabled) {
+            return;
+        }
+
+        super.clickHandler(e);
+
+        if (this.open) {
+            const captured = (e.target as HTMLElement).closest(
+                `[role=${OptionRole.option}]`
+            ) as Option;
+            if (captured && captured.disabled) {
+                return;
+            }
+        }
+
+        this.open = !this.open;
+
+        return true;
+    }
+
+    public focusoutHandler(e: FocusEvent): boolean | void {
+        if (!this.open) {
+            return true;
+        }
+
+        const focusTarget = e.relatedTarget as HTMLElement;
+        if (this.isSameNode(focusTarget)) {
+            DOM.queueUpdate(() => this.focus());
+            return;
+        }
+
+        if (!this.options.includes(focusTarget as Option)) {
+            this.open = false;
+        }
+    }
+
+    public keydownHandler(e: KeyboardEvent): boolean | void {
+        super.keydownHandler(e);
+
+        const keyCode: string = e.key || `${e.key.charCodeAt(0)}`;
+
+        if (keyCode === " " || keyCode === "Enter") {
+            this.open = !this.open;
+        }
+
+        if (keyCode === "Escape") {
+            this.open = false;
+        }
+
+        return true;
+    }
+
+    @observable
+    public value: string;
+
+    public connectedCallback() {
+        super.connectedCallback();
+        DOM.queueUpdate(() => {
+            if (this.proxy instanceof HTMLElement) {
+                this.options.forEach(o => {
+                    const opt = o.proxy;
+                    this.proxy.add(opt);
+                });
+            }
+        });
+    }
+}
+
+/**
+ * Handles proxy element interaction for form association.
+ *
+ * @public
+ */
+class FormAssociatedSelect extends FormAssociated<HTMLSelectElement> {
+    protected proxy: HTMLSelectElement = document.createElement("select");
+
+    protected updateForm(previous, next): void {
+        const value = this.value;
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.value = value;
+        }
+    }
+}
```

Comment:
I'm a little specious that using a mixin approach with FormAssociated won't work here because the mixin code copies prototype methods and in the process will likely override relevant methods on the FormAssociated (constructor, connectedCallback, disconnectedCallback, etc).

Can you confirm that those methods are called when the Select attaches?

Reply:
There is another mixin model I just learned about that could resolve that issue potentially if the FormAssociated lifecycle hooks aren't called.